crates/re_renderer/src/resource_managers/texture_manager.rs

```diff
@@ -116,6 +127,21 @@ impl TextureManager2D {
         Self::create_and_upload_texture(&self.device, &self.queue, texture_pool, creation_desc)
     }
 
+    /// Creates a new 2D texture resource and schedules data upload to the GPU if a texture
+    /// wasn't already created using the same key.
```

Comment:
This need a warning that the `key` MUST be high-entropy (e.g. a hash), or we might get extreme slow-downs